app/models/similarity/unit_similarity_module.rb

```diff
@@ -14,17 +14,26 @@ def last_plagarism_scan
   end
 
   # Pass tasks on to plagarism detection software and setup links between students
-  def check_moss_similarity(force: false)
+  def check_similarity(force: false)
     # Get each task...
     return unless active
 
     # need pwd to restore after cding into submission folder (so the files do not have full path)
     pwd = FileUtils.pwd
 
+    # making temp directory for unit - jplag
+    root_work_dir = Rails.root.join("tmp", "jplag", "#{code}-#{id}")
+    unit_code = "#{code}-#{id}"
+    FileUtils.mkdir_p(root_work_dir)
+
     begin
       logger.info "Checking plagiarsm for unit #{code} - #{name} (id=#{id})"
 
       task_definitions.each do |td|
+        # making temp directory for each task - jplag
+        tasks_dir = root_work_dir.join(td.id.to_s)
+        FileUtils.mkdir_p(tasks_dir)
+
         next if td.moss_language.nil? || td.upload_requirements.nil? || td.upload_requirements.select { |upreq| upreq['type'] == 'code' && upreq['tii_check'] }.empty?
```

Comment:
should this still be moss_language?

Reply:
I'm using the upload requirement language from the task definition ui `task-definition-upload.component.html`.
Feel like it makes sense to use existing language definition considering the task creator has already decided if plagiarism checks are necessary from MOSS.
So if no requirement for MOSS, no requirement for JPLAG. 
Let me know if you don't think is this appropriate.

Reply:
With this we should remove the old moss code, so these could be renamed...